tests/lib/rules/no-multi-spaces.js

```diff
@@ -0,0 +1,136 @@
+  ],
+  invalid: [
+    {
+      code: '<template><div     /></template>',
+      output: '<template><div /></template>',
+      errors: [{
+        message: 'Extra whitespace detected.',
+        type: 'HTMLSelfClosingTagClose'
+      }]
+    },
+    {
+      code: '<template><div   class="foo"  /></template>',
+      output: '<template><div class="foo" /></template>',
+      errors: [
+        {
+          message: 'Extra whitespace detected.',
+          type: 'HTMLIdentifier'
+        },
+        {
+          message: 'Extra whitespace detected.',
+          type: 'HTMLSelfClosingTagClose'
+        }
+      ]
+    },
+    {
+      code: '<template><div\t\tclass="foo"\t\t/></template>',
+      output: '<template><div\tclass="foo"\t/></template>',
+      errors: [
+        {
+          message: 'Extra whitespace detected.',
+          type: 'HTMLIdentifier'
+        },
+        {
+          message: 'Extra whitespace detected.',
+          type: 'HTMLSelfClosingTagClose'
+        }
+      ]
+    },
+    {
+      code: '<template><div   :class="foo"  /></template>',
+      output: '<template><div :class="foo" /></template>',
+      errors: [
+        {
+          message: 'Extra whitespace detected.',
+          type: 'HTMLIdentifier'
+        },
+        {
+          message: 'Extra whitespace detected.',
+          type: 'HTMLSelfClosingTagClose'
+        }
+      ]
+    },
+    {
+      code: '<template><div :foo="" class="foo"  /></template>',
+      output: '<template><div :foo="" class="foo" /></template>',
+      errors: [{
+        message: 'Extra whitespace detected.',
+        type: 'HTMLSelfClosingTagClose'
+      }]
+    },
+    {
+      code: '<template><div foo="" class="foo"  /></template>',
+      output: '<template><div foo="" class="foo" /></template>',
+      errors: [{
+        message: 'Extra whitespace detected.',
+        type: 'HTMLSelfClosingTagClose'
+      }]
+    },
+    {
+      code: '<template><foo v-foo="" class="foo"  /></template>',
+      output: '<template><foo v-foo="" class="foo" /></template>',
+      errors: [{
+        message: 'Extra whitespace detected.',
+        type: 'HTMLSelfClosingTagClose'
+      }]
+    },
+    {
+      code: '<template><foo v-foo="" \n         class="foo"    /></template>',
+      output: '<template><foo v-foo=""\n         class="foo" /></template>',
+      errors: [
+        {
+          message: 'Extra whitespace detected.',
+          type: 'HTMLIdentifier'
+        },
+        {
+          message: 'Extra whitespace detected.',
+          type: 'HTMLSelfClosingTagClose'
+        }
+      ]
+    },
+    {
+      code: '<template><div class="foo "   class="    foo    " /></template>',
+      output: '<template><div class="foo " class="    foo    " /></template>',
+      errors: [
+        {
+          message: 'Extra whitespace detected.',
+          type: 'HTMLIdentifier'
+        }
+      ]
+    }
```

Comment:
I'd like to see tests for the expression of Vue.js directives and mustaches. Those are parsed by script parser (`espree` by default), so those have valid tokens. This rule can report on multiple spaces which are in the expressions.